cabal-testsuite/cabal-testsuite.cabal

```diff
@@ -26,7 +26,7 @@ common shared
   default-language: Haskell2010
 
   build-depends:
-    , base >= 4.6 && <4.18
```

Comment:
Shouldn't we conform this to `<5` as in the other cases?
Especially in the testsuite we do _not_ need _more_ restrictive bounds than in the shipped products.
(Not really topic of this PR, but while we are at it...)